aggregator/aggregator.go

```diff
@@ -162,12 +167,14 @@ func (a *Aggregator) setKey() string {
 
 func (a *Aggregator) AddOrCreate(key string, ts uint32, quantized uint, value float64) {
 	rangeTracker.Sample(ts)
-	aggByKey, ok := a.aggregations[quantized]
+	agg, ok := a.aggregations[quantized]
 	var proc Processor
 	if ok {
-		proc, ok = aggByKey[key]
+		proc, ok = agg.state[key]
 		if ok {
 			// if both levels exist, we can just add the value and that's it
+			agg.count++
+			a.aggregations[quantized] = agg
 			proc.Add(value, ts)
 		}
 	} else {
```

Comment:
seems like this `else` should be removed.
this function has
```
if ok {
...
} else {
...
}
if !ok {
...
}
```
it should be cleaned up to just be
```
if ok {
...
  return
} 
...
```

Reply:
note that the first if has an inner branch in which we we reassign ok again, which is what we check later on.

Reply:
well, if you just return after `proc.Add(value, ts)` you dont need the second `if !ok {` as ok will always be false.